### PR TITLE
Fix alter formatting (minor)

### LINKED ISSUE #48056: clickhouse-format formatting issues (minor)
`clickhouse-format --oneline --query "ALTER TABLE alter_test COMMENT COLUMN IF EXISTS ToDrop 'new comment'"`

should output one line:

`ALTER TABLE alter_test COMMENT COLUMN IF EXISTS ToDrop 'new comment'`

but outputs two lines:

```
ALTER TABLE alter_test
    COMMENT COLUMN IF EXISTS ToDrop 'new comment'
```

The issue is found when testing the #45649 refactoring.

I will be fixing the issue in a separate PR.

### PR DESCRIPTION
**Pull request by murfel**
Fixes half of #48056.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix alter query formatting: if --oneline is specified, format it in one line.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/ASTAlterQuery.cpp [hunk 1]
```diff
@@ -612,7 +612,10 @@ void ASTAlterQuery::formatQueryImpl(const FormatSettings & settings, FormatState
     FormatStateStacked frame_nested = frame;
     frame_nested.need_parens = false;
     frame_nested.expression_list_always_start_on_new_line = true;
-    static_cast<ASTExpressionList *>(command_list)->formatImplMultiline(settings, state, frame_nested);
+    frame_nested.expression_list_prepend_whitespace = true;
```
Comment (vitlibar) [inline src/Parsers/ASTAlterQuery.cpp:615]: Is it necessary if `settings.one_line` is false?

Comment (murfel) [reply src/Parsers/ASTAlterQuery.cpp]: fixed

**Review by alexey-milovidov — APPROVED** [review]

**Review by vitlibar — APPROVED** [review]

File: src/Parsers/ASTAlterQuery.cpp [hunk 2]
```diff
@@ -611,8 +611,16 @@ void ASTAlterQuery::formatQueryImpl(const FormatSettings & settings, FormatState
 
     FormatStateStacked frame_nested = frame;
     frame_nested.need_parens = false;
-    frame_nested.expression_list_always_start_on_new_line = true;
-    static_cast<ASTExpressionList *>(command_list)->formatImplMultiline(settings, state, frame_nested);
+    if (settings.one_line)
+    {
+        frame_nested.expression_list_prepend_whitespace = true;
```
Comment (vitlibar) [inline src/Parsers/ASTAlterQuery.cpp:616]: `frame_nested.expression_list_always_start_on_new_line = settings.one_line` is shorter, but it doesn't really matter

**Comment by vitlibar** [comment]
[test_keeper_map/test.py::test_keeper_map_without_zk](https://s3.amazonaws.com/clickhouse-test-reports/48289/644d83653d4a1b520b6e43bf504c47904cf89766/integration_tests__asan__[1/6].html) - the test failure is known and not related to this PR (https://github.com/ClickHouse/ClickHouse/issues/48181)

**Pull request merged**